### PR TITLE
added async func time execution util

### PR DESCRIPTION
**Pull request by dperetz1**

### CODE REVIEW

**Comment by dperetz1** [comment]
@eitanhs , @brightsource-il/fresco  - I can't add reviewers  (it's a fork , I got no permissions to push a branch), please take a look

File: Bsii.Dotnet.Utils/AsyncUtils.cs [hunk 1]
```diff
@@ -193,6 +194,30 @@ public static async Task<T> TimeoutAfter<T>(this Task<T> task, TimeSpan time, Ca
...
             }
             throw new TimeoutException();
+        }
+
+        /// <summary>
+        /// A shorthand for measuring task execution time 
+        /// <param name="action">The action to be executed</param>
+        /// <returns> a tuple includes a TimeSpan represents the execution time and the result </returns>
+        /// </summary>
+        public static async Task<(T res, TimeSpan elapsed)> TimeExecutionAsync<T>(this Func<Task<T>> action)
```
Comment (kpahima) [inline Bsii.Dotnet.Utils/AsyncUtils.cs:204]: Please add the SFC synch functions as well so SFC can use this Nuget as well

**Review by kpahima — CHANGES_REQUESTED** [review]

File: Bsii.Dotnet.Utils/AsyncUtils.cs (hunk 1 above)
Comment (dperetz1) [reply Bsii.Dotnet.Utils/AsyncUtils.cs]: done

**Review by kpahima — DISMISSED** [review]

**Review by eitanhs — COMMENTED** [review]
Thanks!
Please add unit tests...

**Comment by dperetz1** [comment]
added unit tests

**Comment by dperetz1** [comment]
@eitanhs ping

**Review by eitanhs — APPROVED** [review]

**Review by eitanhs — APPROVED** [review]

**Pull request merged**